src/EventListener/AddEnabledLocalesListener.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\EventListener;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\Routing\RequestContextAwareInterface;
+
+/**
+ * Choose the locale to use according to the Accept-Language header
+ *
+ * @author Grégoire Hébert <contact@gheb.dev>
+ */
+final class AddEnabledLocalesListener
+{
+    private $defaultEnabledLocales;
+    private $router;
+    private $resourceMetadataFactory;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, RequestContextAwareInterface $router = null, array $defaultEnabledLocales = [])
```

Comment:
In my opinion, this listener belongs in Symfony. /cc @nicolas-grekas

Reply:
this has been rejected in https://github.com/symfony/symfony/pull/36507, isn't it?
(the arguments against it would apply to this PR btw ;) )

Reply:
I don't understand the arguments against. It's opt-in (so it cannot be a security vulnerability, the linked one is not related), it's a core feature of HTTP and it's very useful for APIs.